addons/mail/static/src/core/thread_model.js

```diff
@@ -415,4 +415,8 @@ export class Thread {
         }
         return this._store.messages[Math.max(...previousMessages.map((m) => m.id))];
     }
+
+    get pinnedMessages() {
+        return this.messages.filter(({ pinned }) => pinned);
+    }
```

Comment:
Showing of pinned messages should be ordered based on python relational field `pinned_message_ids`.

Would be best to handle some kind of `fetch(New/More)Messages` architecture in the display of pinned message.

Reply:
I plan to fetch directly all pinned messages when opening the pinned panel (actually there could be some limit but right now only loaded messages display in the pinned panel). But this requires an heavy refactoring of the current message fetch architecture because then, we will have the knowledge of several message ranges without knowing if/what's in between. In plan to do it on a follow up PR since it is quiet a lot of work. The idea would be to do in a fashion similar to the one discord use:

- initial fetch
- jump to far away message --> fetch messages `around`
- scroll up --> fetch messages `before` 
- scroll down --> fetch messages `after`
- keep track of the known message range to know whether they are messages in between or if they are yet to load.